tests/products/test_file.py

```diff
@@ -301,25 +301,28 @@ def test_task_without_client_is_outdated_by_code(tmp_directory):
     assert t.product._is_outdated()
 
 
-def _edit_source_code(path):
+def _edit_source_code(path, tmp_path):
```

Comment:
why do we need this tmp_path argument?

Reply:
Was trying to write the json to a temporary path and then replace it with the original filename. That seemed to solve the issue that was coming up earlier.